xds/src/main/java/io/grpc/xds/DropStatsCounter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 20121 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Stopwatch;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Recorder for dropped requests.
+ */
+// TODO(chengyuanzhang): this class can be moved into LoadStatsManager.
+@ThreadSafe
+final class DropStatsCounter {
+
+  private final AtomicLong uncategorizedDrops = new AtomicLong();
+  private volatile ConcurrentMap<String, AtomicLong> categorizedDrops = new ConcurrentHashMap<>();
```

Comment:
There won't be many drop categories, typically two, so clear up the map does not provide benefit. A final map will be much cleaner.

Reply:
> There won't be many drop categories, typically two

Shall we make such an assumption? I am not sure, as currently drops aren't even used at all.

If we do not clean up entries in the map in `snapshot()`, following `snapshot()` calls will always include zero-count entries, such as `"lb": 0`, etc. It may be acceptable. But replacing the map in `snapshot()` isn't too complicated right?

Reply:
Just calling `categorizedDrops.clear()` is much simpler than replacing the map. The race for missing one drop record is still there. `clear()` does not shrink the allocation but the number of keys won't be unbounded, so replacing the map does not give much memory gain.

Reply:
Alright, changed to use `clear()`.